src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { EntryType } from 'app/shared/enum/entry-type';
+import { EntryUpdateTime, User, UsersService } from 'app/shared/openapi';
+import { UserQuery } from 'app/shared/user/user.query';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { WorkflowService } from 'app/shared/state/workflow.service';
+import { WorkflowQuery } from 'app/shared/state/workflow.query';
+import { MyEntriesQuery } from 'app/shared/state/my-entries.query';
+import { SessionQuery } from 'app/shared/session/session.query';
+import { SessionService } from 'app/shared/session/session.service';
+import { MyWorkflowsService } from 'app/myworkflows/myworkflows.service';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+
+@Component({
+  selector: 'app-entry-box',
+  templateUrl: './entry-box.component.html',
+  styleUrls: ['./entry-box.component.scss'],
+})
+export class EntryBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  @Input() entryType: EntryUpdateTime.EntryTypeEnum;
+  entryTypeLowerCase: string;
+  entryTypeCapitalize: string;
+  filterText: string;
+  listOfEntries: Array<EntryUpdateTime> = [];
+  user: User;
+  helpLink: string;
+  allEntriesLink: string;
+  public isLoading = true;
+
+  constructor(
+    private registerToolService: RegisterToolService,
+    protected userQuery: UserQuery,
+    protected workflowService: WorkflowService,
+    protected workflowQuery: WorkflowQuery,
+    protected myWorkflowsService: MyWorkflowsService,
+    protected myEntriesQuery: MyEntriesQuery,
+    protected sessionQuery: SessionQuery,
+    protected sessionService: SessionService,
+    private usersService: UsersService
+  ) {
+    super();
+    this.user = this.userQuery.getValue().user;
+  }
+
+  ngOnInit(): void {
+    this.getMyEntries();
+    if (this.entryType) {
+      this.entryTypeLowerCase = this.entryType.toLowerCase();
+      this.entryTypeCapitalize = this.entryTypeLowerCase[0].toUpperCase() + this.entryTypeLowerCase.substring(1);
```

Comment:
By changing the `@Input() entryType` from the string, I did have to do this clunky changing back into a capitalized format, is there a more efficient way? I tried to use `text-transform` but that made the UI look wonky unless more styles are added which may be a bit more work than this option. The capitalized version does make specific elements look nicer so ideally I didn't want to use the lowercase version.

Reply:
Note that it is really only used in 3 places (Header title, Register a ___ button, and All ___ link at the bottom)

Reply:
There is an Angular [title case pipe](https://angular.io/api/common/TitleCasePipe) you can use, either in the .ts file or in the HTML.

Sorry, I know it's sometimes a bit of a hassle to have to use a type instead of just free text, but that's how one is generally supposed to use typed languages like TypeScript and Java, to let the compiler catch any errors.

Reply:
Ohh that is something I didn't know but is useful to know, it also makes this a LOT cleaner and simpler. Having it as a type instead of a string makes sense to me especially to prevent errors, thank you! :clap: